photon-ml/src/main/scala/com/linkedin/photon/ml/model/Coefficients.scala

```diff
@@ -145,10 +147,13 @@ protected[ml] object Coefficients {
    */
   def apply(length: Int)(indices: Int*)(nnz: Double*): Coefficients = {
 
-    require(0 < length)
-    require(indices.length == nnz.length)
-    require(indices.sorted == indices)
-    // TODO: check for duplicates?
+    require(0 < length, s"Specified length < 0 for Coefficients: $length")
+    require(indices.length == nnz.length,
+      s"""There are ${indices.length} indices but ${nnz.length} non-zero specified for Coefficients.
+         | There should be as many indices as non-zero values.
+       """.stripMargin)
+    require(indices.sorted == indices, s"Indices need to be in strictly increasing order for Coefficients")
```

Comment:
According to me, the check in VectorUtils is too distant from the actual object that contains the coefficients, which makes it easy to forget/not use that function in the future. If we _localize _ that check in the Coefficients constructor, even when we refactor, or no longer go though the function in VectorUtils, we'll still check. On balance, I would vote to move the check to be in the constructor of the object.